dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasResource.java

```diff
@@ -0,0 +1,91 @@
+
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+import com.codahale.metrics.annotation.Timed;
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.DockstoreWebserviceConfiguration;
+import io.dockstore.webservice.core.User;
+import io.dockstore.webservice.core.WorkflowVersion;
+import io.dockstore.webservice.jdbi.WorkflowDAO;
+import io.dockstore.webservice.jdbi.WorkflowVersionDAO;
+import io.dropwizard.auth.Auth;
+import io.dropwizard.hibernate.UnitOfWork;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.Authorization;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.hibernate.SessionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static io.dockstore.webservice.Constants.JWT_SECURITY_DEFINITION_NAME;
+
+@Path("/aliases")
+@Api("/aliases")
+public class AliasResource {
+
+    private static final String OPTIONAL_AUTH_MESSAGE = "Does not require authentication for published workflows,"
+            + " authentication can be provided for restricted workflows";
+
+    private static final Logger LOG = LoggerFactory.getLogger(AliasResource.class);
+    protected final WorkflowVersionDAO workflowVersionDAO;
+    protected final WorkflowDAO workflowDAO;
+    private final WorkflowResource workflowResource;
+    private final WorkflowVersionResource workflowVersionResource;
+
+
+    public AliasResource(HttpClient client, SessionFactory sessionFactory, WorkflowResource workflowResource,
+            WorkflowVersionResource workflowVersionResource, DockstoreWebserviceConfiguration configuration) {
+        this.workflowVersionDAO = new WorkflowVersionDAO(sessionFactory);
+        this.workflowDAO = new WorkflowDAO(sessionFactory);
+        this.workflowResource = workflowResource;
+        this.workflowVersionResource = workflowVersionResource;
+    }
+
+
+    @POST
+    @Timed
+    @UnitOfWork
+    @Path("workflow-versions/{workflowVersionId}")
+    @ApiOperation(nickname = "addAliases", value = "Add aliases linked to a workflow version in Dockstore.", authorizations = {
+            @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, notes = "Aliases are alphanumerical (case-insensitive "
+            + "and may contain internal hyphens), given in a comma-delimited list.", response = WorkflowVersion.class)
+    public WorkflowVersion addAliases(@ApiParam(hidden = true) @Auth User user,
+            @ApiParam(value = "workflow version to modify.", required = true) @PathParam("workflowVersionId") Long workflowVersionId,
+            @ApiParam(value = "Comma-delimited list of aliases.", required = true) @QueryParam("aliases") String aliases) {
+        return workflowVersionResource.addAliases(user, workflowVersionId, aliases);
```

Comment:
You need to validate that the workflowVersionId exists.

Reply:
If you require the workflow id on the path, then it's easier to validate that the version exists and that the user has rights to create an alias (are there limitations on that -- can I create an alias for any workflow or just mine?). Then you could also store the workflow id in the alias table. That will make lookup easier.

If you don't require the workflow id, then you will still need to look up the workflow for validation. If you go this route, I also suggest you store the workflow id in the alias table.

Reply:
AliasResource.java POST -> WorkfowVersionResource.addAliases -> AliaseableResourceInterface.addAliasesAndCheck -> WorkflowVersionResource#getAndCheckResource -> workflowVersionDAO.findById (throws exception if it is not found) -> workflowResource.checkEntry (checks if the Workflow exists) -> workflowResource.checkUserCanUpdate(user, workflow) -> AuthenticatedResourceInterface.checkUser -> AuthenticatedResourceInterface.userCannotRead (checks if user is in Entry's user list)

Reply:
Did not add workflow id to the alias table. I leave that for another PR.